File: RxTools-library/src/main/java/com/vondear/rxtools/view/wheelhorizontal/OnWheelChangedListener.java
package com.vondear.rxtools.view.wheelhorizontal;


/**
 * @author vondear
 * Wheel changed listener interface.
 * <p>The onChanged() method is called whenever current spinnerwheel positions is changed:
 * <li> New Wheel position is set
 * <li> Wheel view is scrolled
 */
public interface OnWheelChangedListener {
	/**
	 * Callback method to be invoked when current item changed
	 * @param wheel the spinnerwheel view whose state has changed
	 * @param oldValue the old value of current item
	 * @param newValue the new value of current item
	 */
	void onChanged(AbstractWheel wheel, int oldValue, int newValue);
}

File: RxUI/src/main/java/com/tamsiree/rxui/view/dialog/wheel/OnWheelChangedListener.kt
package com.tamsiree.rxui.view.dialog.wheel

/**
 * @author tamsiree
 * Wheel changed listener interface.
 *
 * The onChanged() method is called whenever current wheel positions is changed:
 *  *  New Wheel position is set
 *  *  Wheel view is scrolled
 */
interface OnWheelChangedListener {
    /**
     * Callback method to be invoked when current item changed
     * @param wheel the wheel view whose state has changed
     * @param oldValue the old value of current item
     * @param newValue the new value of current item
     */
    fun onChanged(wheel: WheelView, oldValue: Int, newValue: Int)
}